x-pack/platform/plugins/shared/alerting/common/routes/rule_template/apis/find/schemas/v1.ts

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { schema } from '@kbn/config-schema';
+import { stringOrStringArraySchema } from '../../../../../schemas';
+
+export const findRuleTemplatesRequestQuerySchema = schema.object({
+  per_page: schema.number({
+    defaultValue: 10,
+    min: 0,
+    meta: {
+      description: 'The number of rule templates to return per page.',
+    },
+  }),
+  page: schema.number({
+    defaultValue: 1,
+    min: 1,
+    meta: {
+      description: 'The page number to return.',
+    },
+  }),
+  search: schema.maybe(
+    schema.string({
+      meta: {
+        description:
+          'An Elasticsearch simple_query_string query that filters the objects in the response.',
+      },
+    })
+  ),
+  default_search_operator: schema.maybe(
+    schema.oneOf([schema.literal('OR'), schema.literal('AND')], {
+      defaultValue: 'OR',
+      meta: {
+        description: 'The default operator to use for the simple_query_string.',
+      },
+    })
+  ),
+  sort_field: schema.maybe(
```

Comment:
I would say better to limit it to a known list of attributes. Like `title`, etc without exposing any implementation details like knowing the exact name of the attributes in the mappings. Then we can do any transformation we need on the rule client. If the field is not needed it we can remove it and default internally to sort by title or creation date.

Reply:
would that look like e.g. `schema.oneOf([schema.literal('name'), schema.literal('tags')]` ?